Source/WebKit/WebProcess/WebPage/Cocoa/WebPageCocoa.mm

```diff
@@ -156,6 +156,10 @@
         && (MIMETypeRegistry::isPDFOrPostScriptMIMEType(contentType)
             || (contentType.isEmpty()
                 && (path.endsWithIgnoringASCIICase(".pdf"_s) || path.endsWithIgnoringASCIICase(".ps"_s))));
+#if ENABLE(UNIFIED_PDF)
+    shouldUsePlugin = shouldUsePlugin || corePage()->settings().unifiedPDFEnabled();
```

Comment:
You still need the MIME type checks don't you?

Reply:
Changed so that we no longer skip over the type checking https://github.com/WebKit/WebKit/commit/34f1fc9ef36a4c6d5f5e333cf00080db7b704674